### PR TITLE
fix: Tweak survey settings tab and share modal to shorten viral loop

### PR DESCRIPTION
**Pull request by jobenjada**
## What does this PR do?

1. Tweak "not connected" banner
2. Streamline settings tab for Link Surveys
3. Streamline Share Modal

**Tweaked banner before and after:**
image

image

**Streamline Settings Tab:**
- Changed which cards are default open
- Completely hide (instead of disabled) irrelevant settings if type link survey
- Revamped Styling card: Remove toggle for settings which cannot be overwritten because there is no global value
- Added hint to global settings

@mattinannt there is unused code bc I removed the toggle. I'd just remove it or am I missing something?
image

@Dhruwang as discussed in Slack, pls add instant updating of global settings when changed and survey editor is open

**Streamline Share Modal**

On Share Survey this modal is shown, to focus on sharing the link around. It then takes the user deeper into the product (alerts / integrations):

image

### CODE REVIEW

File: apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/(analysis)/summary/components/ShareEmbedSurvey.tsx [hunk 1]
```diff
@@ -22,91 +26,169 @@ interface ShareEmbedSurveyProps {
...
+  environmentId: string;
 }
 export default function ShareEmbedSurvey({
   survey,
   open,
   setOpen,
   webAppUrl,
-  product,
   user,
+  environmentId,
```
Comment (mattinannt) [inline apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/(analysis)/summary/components/ShareEmbedSurvey.tsx:37]: You can get the `environmentId` from the survey. No need to pass it down 😊

File: apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/SurveyEditor.tsx [hunk 2]
```diff
@@ -42,7 +43,7 @@ export default function SurveyEditor({
   const [activeQuestionId, setActiveQuestionId] = useState<string | null>(null);
   const [localSurvey, setLocalSurvey] = useState<TSurvey | null>();
   const [invalidQuestions, setInvalidQuestions] = useState<String[] | null>(null);
-
+  const [product, setproduct] = useState<TProduct>(productInfo);
```
Comment (mattinannt) [inline apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/SurveyEditor.tsx:46]: I think we should keep it like we did with the survey:
Pass `product` as a prop and rename the product in the state to `localProduct`. It still might not be ideal but at least this way it's consistent.

File: apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/SurveyEditor.tsx [hunk 3]
```diff
@@ -54,6 +55,24 @@ export default function SurveyEditor({
...
+          }
+        };
+        fetchLatestProduct();
+      }
+    };
+    document.addEventListener("visibilitychange", listener);
+    return () => {
+      document.removeEventListener("visibilitychange", listener);
+    };
+  }, []);
```
Comment (mattinannt) [inline apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/SurveyEditor.tsx:74]: getting this build warning:
```
@formbricks/web:build: ./app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/SurveyEditor.tsx
@formbricks/web:build: 74:6  Warning: React Hook useEffect has a missing dependency: 'product.id'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps
```

**Review by mattinannt — COMMENTED** [review]
@Dhruwang @jobenjada thanks for this big tweak. I haven't tested the functionality and frontend work fully (did you @jobenjada  ?), just fixed the build errors and only have a few change requests while going through the code:

**Review by mattinannt — APPROVED** [review]
@Dhruwang looks great; thank you for the changes :-) 💪

**Review by Dhruwang — COMMENTED** [review]
Hey @jobenjada , tweaked survey share modal looks pretty dope 🚀😊

Here are 2 issues I have noticed:

1. Cannot scroll in `embed in an email tab`

Screenshot 2024-01-16 at 3 06 52 PM

2. Breaks for single use link surveys


https://github.com/formbricks/formbricks/assets/67850763/90fa6c87-7377-4315-be69-0a3fc303ece4

Also after creating a single use link survey, we get a link on survey share modal that is invalid because of absence of suid

**Review by mattinannt — APPROVED** [review]
@Dhruwang thank you for the changes; looks great! 😊
Let's merge this 💪🚀

**Pull request merged**